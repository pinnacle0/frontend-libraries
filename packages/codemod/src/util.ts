import path from "path";
import * as t from "@babel/types";
import glob from "glob";
import {parse} from "@babel/parser";
import traverse from "@babel/traverse";
import generate from "@babel/generator";
import type {API, Transform} from "./type";
import type {ModType} from "./modType";

export function createApi(): API {
    return {
        parse: (source: string) =>
            parse(source, {
                sourceType: "module",
                attachComment: true,
                plugins: ["jsx", "typescript", "decorators-legacy"],
            }),
        builder: t,
        traverse,
        generate,
    };
}

export function resolveCodemod(type: ModType): Transform | null {
    try {
        const modPath = require.resolve(path.join(__dirname, "./mod/" + type));
        return require(modPath).default;
    } catch (e) {
        return null;
    }
}

export function globPromise(path: string, options: glob.IOptions): Promise<string[]> {
    return new Promise<string[]>((resolve, reject) => {
        glob(path, options, (error, matches) => {
            if (error) {
                reject(error);
            } else {
                resolve(matches);
            }
        });
    });
}
