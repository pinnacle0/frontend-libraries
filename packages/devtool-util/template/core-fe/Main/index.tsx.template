import React from "react";
import {{1}Module} from "../module"
import "./index.less";

export const Main: React.ComponentType = {1}Module.attachLifecycle(() => {
    return <div />;
});
