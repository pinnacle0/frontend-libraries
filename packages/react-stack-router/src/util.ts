const PREFIX = "[react-stack-router]";

export function invariant(condition: any, message: string): asserts condition {
    if (condition) {
        return;
    }
    throw new Error(`${PREFIX} ${message}`);
}

export function createKey() {
    return Date.now().toString(36) + Math.random().toString(36).substring(2);
}
