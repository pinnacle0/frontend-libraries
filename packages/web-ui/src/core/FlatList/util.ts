import {Direction} from "../../hooks/useSwipe";
import type {Boundary} from "./type";

const OVER_BOUNDARY_LIMIT = 40;

export function isExceededBounary(delta: number, boundary: Boundary): boolean {
    switch (boundary) {
        case "top":
            return delta < -OVER_BOUNDARY_LIMIT;
        case "bottom":
            return delta > OVER_BOUNDARY_LIMIT;
    }
}

export function getBounaryFromStartDirection(direction: Direction): Boundary | null {
    switch (direction) {
        case Direction.DOWN:
            return "top";
        case Direction.UP:
            return "bottom";
        default:
            return null;
    }
}

export function isScrollTop(element: HTMLElement): boolean {
    return element.scrollTop === 0;
}

export function isScrollBottom(element: HTMLElement): boolean {
    const {scrollTop, clientHeight, scrollHeight} = element;
    return Math.ceil(scrollTop) + clientHeight >= scrollHeight;
}

export function isScrollLeft(element: HTMLElement) {
    return element.scrollLeft === 0;
}

export function isScrollRight(element: HTMLElement): boolean {
    const {scrollWidth, scrollLeft, clientWidth} = element;
    return Math.ceil(scrollLeft) + clientWidth >= scrollWidth;
}
