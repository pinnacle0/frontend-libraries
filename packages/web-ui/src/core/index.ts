export {type Props as CarouselProps} from "./Carousel";
export {Carousel} from "./Carousel";
export {type Props as YearMonthSelectorProps} from "./YearMonthSelector";
export {YearMonthSelector} from "./YearMonthSelector";
export {type Props as RelativeTimeProps} from "./RelativeTime";
export {RelativeTime} from "./RelativeTime";
export {type Props as WithTooltipListProps} from "./WithTooltipList";
export {WithTooltipList} from "./WithTooltipList";
export {type Props as ButtonProps} from "./Button";
export {Button} from "./Button";
export {type Props as TimeRangePickerProps} from "./TimeRangePicker";
export {TimeRangePicker} from "./TimeRangePicker";
export {type Props as SpinProps} from "./Spin";
export {Spin} from "./Spin";
export {type Props as CardProps} from "./Card";
export {Card} from "./Card";
export {type Props as RadioProps} from "./Radio";
export {Radio} from "./Radio";
export {type Props as SkeletonProps} from "./Skeleton";
export {Skeleton} from "./Skeleton";
export {type Props as AmountRangeInputProps} from "./AmountRangeInput";
export {AmountRangeInput} from "./AmountRangeInput";
export {type Props as AutoCompleteInputProps} from "./AutoCompleteInput";
export {AutoCompleteInput} from "./AutoCompleteInput";
export {type Props as OrientationContextProps} from "./OrientationContext";
export {OrientationContextProvider} from "./OrientationContext";
export {type Props as BadgeProps} from "./Badge";
export {Badge} from "./Badge";
export {type Props as DateRangePickerProps} from "./DateRangePicker";
export {DateRangePicker} from "./DateRangePicker";
export {type Props as BoolSwitchProps} from "./BoolSwitch";
export {BoolSwitch} from "./BoolSwitch";
export {type Props as LabelledSelectProps} from "./LabelledSelect";
export {LabelledSelect} from "./LabelledSelect";
export {Progress} from "./Progress";
export {type Props as TooltipProps} from "./Tooltip";
export {Tooltip} from "./Tooltip";
export {type Props as SignedAmountInputProps} from "./SignedAmountInput";
export {SignedAmountInput} from "./SignedAmountInput";
export {type Props as DocumentTitleProps} from "./DocumentTitle";
export {DocumentTitle} from "./DocumentTitle";
export {Grid} from "./Grid";
export {EnumSelect} from "./EnumSelect";
export {type Props as EnumSelectProps} from "./EnumSelect";
export {type Props as BoolRadioProps} from "./BoolRadio";
export {BoolRadio} from "./BoolRadio";
export {type Props as BoolSelectProps} from "./BoolSelect";
export {BoolSelect} from "./BoolSelect";
export {type Props as InputProps} from "./Input";
export {Input} from "./Input";
export {type Props as ImageProps} from "./Image";
export {Image} from "./Image";
export {type Props as MultipleEnumSelectProps} from "./MultipleEnumSelect";
export {MultipleEnumSelect} from "./MultipleEnumSelect";
export {type TableProps} from "./Table";
export {Table} from "./Table";
export {type Props as UploaderProps} from "./Uploader";
export {Uploader} from "./Uploader";
export {type Props as LocalImporterProps} from "./LocalImporter";
export {LocalImporter} from "./LocalImporter";
export {type Props as TagsProps} from "./Tags";
export {Tags} from "./Tags";
export {type Props as LinkProps} from "./Link";
export {Link} from "./Link";
export {type Props as TimePickerProps} from "./TimePicker";
export {TimePicker} from "./TimePicker";
export {type Props as SelectProps} from "./Select";
export {Select} from "./Select";
export {type Props as AmountConditionInputProps} from "./AmountConditionInput";
export {AmountConditionInput} from "./AmountConditionInput";
export {type Props as DateCalendarProps} from "./DateCalendar";
export {DateCalendar} from "./DateCalendar";
export {type Props as CheckboxProps} from "./Checkbox";
export {Checkbox} from "./Checkbox";
export {type Props as StepsProps} from "./Steps";
export {Steps} from "./Steps";
export {type Props as DateTimePickerProps} from "./DateTimePicker";
export {DateTimePicker} from "./DateTimePicker";
export {type Props as SearchableSelectProps} from "./SearchableSelect";
export {SearchableSelect} from "./SearchableSelect";
export {type Props as DrawerProps} from "./Drawer";
export {Drawer} from "./Drawer";
export {type Props as TagProps} from "./Tag";
export {Tag} from "./Tag";
export {type Props as DatePickerProps} from "./DatePicker";
export {DatePicker} from "./DatePicker";
export {type Props as LocaleSelectProps} from "./LocaleSelect";
export {LocaleSelect} from "./LocaleSelect";
export {type Props as UIProviderProps} from "./UIProvider";
export {UIProvider} from "./UIProvider";
export {type Props as DropdownProps} from "./Dropdown";
export {Dropdown} from "./Dropdown";
export {type Props as SpaceProps} from "./Space";
export {Space} from "./Space";
export {type Props as CollapseProps} from "./Collapse";
export {Collapse} from "./Collapse";
