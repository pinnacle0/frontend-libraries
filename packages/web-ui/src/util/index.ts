export * from "./NotificationUtil";
export * from "./MessageUtil";
export * from "./ReactUtil";
export * from "./PromptUtil";
export * from "./LocaleUtil";
export * from "./RecentUsedStorageUtil";
export * from "./ExpirableLocalStorageUtil";
export * from "./LocalStorageUtil";
export * from "./UploadUtil";
export * from "./MediaUtil";
export * from "./DayJS";
export * from "./SwipeUtil";
export * from "./SessionStorageUtil";
export * from "./ModalUtil";
export * from "./BrowserUtil";
export * from "./AdminNavigationUtil";
export * from "./CanvasUtil";
export * from "./ClassNames";
export * from "./Clipboard";
export * from "./ExportUtil";
export * from "./ImportUtil";
export * from "./OrientationUtil";
