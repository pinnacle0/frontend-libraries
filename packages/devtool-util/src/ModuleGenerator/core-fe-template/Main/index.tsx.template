import React from "react";
import {ReactUtil} from "@pinnacle0/web-ui/util/ReactUtil";
import "./index.less";

export const Main = ReactUtil.memo("Main", () => {
    return <div />;
});
