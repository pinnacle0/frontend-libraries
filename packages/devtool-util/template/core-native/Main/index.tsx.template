import React from "react";
import {StyleSheet, View} from "react-native";
import {ReactUtil} from "@pinnacle0/web-ui/util/ReactUtil";

export const Main = ReactUtil.memo("Main", () => {
    return (
        <View style={styles.container}>
            <View />
        </View>
    );
});

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
});
