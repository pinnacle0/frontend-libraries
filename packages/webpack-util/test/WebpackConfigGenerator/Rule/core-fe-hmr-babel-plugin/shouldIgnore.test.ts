import * as babel from "@babel/core";
import plugin from "../../../../src/WebpackConfigGenerator/Rule/core-fe-hmr-babel-plugin";

describe("core-fe-hmr-babel-plugin", () => {
    test("should ignore when non core-fe source files", () => {
        const source = `
            import {Module} from "not-core-fe";
            import {Main} from "./component/Main";

            const initialState = {};
            class FeatureModule extends Module {}

            const featureModule = register(new FeatureModule("feature", initialState));
            export const actions = featureModule.getActions();
            export const MainComponent = featureModule.attachLifecycle(Main);
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;
        expect(code).toMatchSnapshot();
        expect(code).not.toContain("if (module.hot) module.hot.decline();");
    });

    test("sanity test for some react component files", () => {
        const source = `
            import React from "react";
            import {ReactUtil} from "@pinnacle0/util";

            export const Main = ReactUtil.memo("Main", () => {
                return React.createElement("div", {});
            });
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;

        expect(code).toMatchSnapshot();
        expect(code).not.toContain("if (module.hot) module.hot.decline();");
    });
});
