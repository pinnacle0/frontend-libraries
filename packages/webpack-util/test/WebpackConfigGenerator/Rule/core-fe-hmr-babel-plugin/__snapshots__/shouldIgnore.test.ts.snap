// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core-fe-hmr-babel-plugin sanity test for some react component files 1`] = `
"import React from \"react\";
import { ReactUtil } from \"@pinnacle0/util\";
export const Main = ReactUtil.memo(\"Main\", () => {
  return React.createElement(\"div\", {});
});"
`;

exports[`core-fe-hmr-babel-plugin should ignore when non core-fe source files 1`] = `
"import { Module } from \"not-core-fe\";
import { Main } from \"./component/Main\";
const initialState = {};

class FeatureModule extends Module {}

const featureModule = register(new FeatureModule(\"feature\", initialState));
export const actions = featureModule.getActions();
export const MainComponent = featureModule.attachLifecycle(Main);"
`;
